### ISSUE TITLE
hyperlinkUnderline causes NumberFormatException

### ISSUE THREAD

**Issue by VladimirAlexiev** [plantuml/plantuml#1411]
From the documentation https://plantuml.com/link#950c6270a3078b08 **hyperlinkUnderline**:
[The example](https://www.plantuml.com/plantuml/uml/SoWkIImgAStDuIhEpimhI2nAp5N8g2n8BSh9pCiEpKi1Cr8LIefAKxciu1BE-JdvHGh5gIKbcNdfrbnMqT4P9IK5Llhw1Jc9UI3Jz9BpSsDZhRaSKlDIG8O30000) causes java.lang.NumberFormatException

**Comment by jimnelson372** [comment]
Interesting.  Looking through the code, I found that other than for the "false" condition, the hyperlinkUnderline skinparm is actually expected to be a number, specifying the underline thickness.   If "false" it is interpreted as 0, but if "true" (as in the example) it gets the exception while trying to interpret "true" as a number.

Giving a number greater than 0 currently works, with >1 working only on the PNG, not the SVG (as far as I can tell).  For example:

```
@startuml
skinparam hyperlinkUnderline 1
:hyperlinkColor setting;
:[[http://plantuml.com]];
@enduml
```
generates:
![image](https://github.com/plantuml/plantuml/assets/66284321/d0d1a3bc-55ce-4502-b889-b56bb7fa5beb)

and:

```
@startuml
skinparam hyperlinkUnderline 5
:hyperlinkColor setting;
:[[http://plantuml.com]];
@enduml
```
generates the PNG of:

![image](https://github.com/plantuml/plantuml/assets/66284321/504685b5-c4d2-4f08-8f86-c5417a2df80a)

It would be a simple fix to make "true" be interpreted as 1, just as "false" is currently interpreted as 0.   Or, in case the default isn't just 1, "true" could just result in the default underline, which happens if this skinparam isn't included at all.

The condition just needs to be added to the FromSkinparamToStyle.convertNow(...) method, where the false condition on "hyperlinkunderline" is converted.

I'm happy to create the pull request for that change.   I've confirmed that the issue exists up through the latest code on github.   I'd just appreciate feedback on whether "true" should convert to 1 or to just the default.

Regards,

Jim N.

**Comment by jimnelson372** [comment]
I should add that I'm not sure if the use of a number value for increasing the thickness of the line to be larger than 1 was intentional.  The documentation doesn't mention this, and as I said the SVG doesn't increase the thickness beyond the default.   So, it's possible the correct fix is to ensure that regardless of the number provided (5 in my example) it should only be set to 1.  That would be consistent with the documentation.

Again, I can make the change, but I'm not clear about which of the 3 interpretations was intended or is most appropriate, given the documentation and whether any users already interpret it as a thickness setting.

1. 'true' would mean 1.
2. 'true' would mean default.
3. Greater than 1 would still just be 1.

Regards,

Jim N

**Comment by VladimirAlexiev** [comment]
@jimnelson372 thanks for debugging this!
I vote to keep the option >1 even if it works in PNG only.
So, treat false as 0 and true as 1.

I've documented it: http://alphadoc.plantuml.com/doc/markdown/en/link#pf36k8k5wf9pl1o3tsvf

**Comment by VladimirAlexiev** [comment]
@jimnelson372 Alas, `hyperlinkUnderline 5` doesn't work in class diagrams, eg see [this diagram](https://www.plantuml.com/plantuml/uml/TOwn2i8m48RtFCNHLTg2uk8W1K5y0QUIuMmFKvhcmYMrBUAxQxFeud8SVF__yLSXeiHni2hqdVCeE837Q2xy83cnPO6x4BaVR1PFOZlNNrsR7eAzQYo606VcBM20hAvDZFvGbjCq5GiQvgBXeKnyFLfdy5G0cvR7cwLQmv_63BFMto0iwgMI9S-FgscdAdBjP_yR)
![image](https://github.com/plantuml/plantuml/assets/536250/27c946ce-400f-4eed-a74c-9beba3ae46d3)

Can you debug this too?

**Comment by jimnelson372** [comment]
You mean in that it doesn't account for thickness.   When "true" it's the same problem as the other diagram, so it will work when true.  As for increased thickness, I can look, but I'm not clear if that's would technically be a but fix or an enhancement.  But, I'll look to see if it's just as simple as making the class diagram use the same few lines of code as the activity diagram uses for thickness.

**Comment by jimnelson372** [comment]
@VladimirAlexiev, I haven't yet had a chance to look further at why the Class underline isn't getting thicker, but I came across a forum discussion where @The-Lum mentioned there is a hyperlinkUnderlineThickness setting.  Here is that forum discussion where he shows how to use it:

https://forum.plantuml.net/17029/how-to-use-skinparam-hyperlinkunderline-false

and he mentions an issue involving its use.

https://github.com/plantuml/plantuml/issues/1241

Unfortunately, settings above 1 on hyperlinkUnderlineThickness seem to be ignored even for the example I gave above showing a thickness of 5 when using hyperlinkUnderline itself with a number.  

While the fix to hyperlinkUnderline would stopped the stack dump on "hyperlinkUnderline true", I think the in light of this hyperlinkUnderlineThickness parameter, work to get more objects to show thicker than 1 underlines should probably go on an issue for that, leaving hyperlinkUnderline itself a true/false parameter.

By the way, my code commit for the fix I mentioned showed up just above this comment.  It's the simple version that would have true mean 1 and not change anything else.  I haven't yet created a pull request.   

Perhaps @The-Lum could share more about the history of the hyperlinkUnderlineThickness compared to hyperlinkUnderline and what the plans were for each of them.

Regards,

Jim Nelson

**Comment by jimnelson372** [comment]
Actually, I think I should go ahead and submit the pull request, making the true in `"skinparam hyperlinkUnderline true"` mean 1.  That way the stack dump you're currently seeing goes away.  Then a new issue could be opened to discuss if this or hyperlinkUnderlineThickness when given values >1 should increase the thickness of those hyperlink underlines across the board, not just on Activities and not just in PNG.   I'm going to look at a few more things this evening, but then I will submit the fix I've already made.

Regards,

Jim Nelson

**Comment by jimnelson372** [comment]
Hi @VladimirAlexiev.  I did debug the class diagram you shared.  I discovered a few things:

1. It turns out that "skinparam **hyperlinkUnderline**" is actually code to effectively set the same parameter as **hyperlinkUnderlineThickness**.   So, for numeric values, both are effectively the same thing.  

2. While a range of entities actually will draw thicker underlines on their hyperlink if they find **hyperlinkUnderlinethickness** is >1, at least for PNG generated diagrams.   In the example of **hyperlinkUnderlineThickness** that was shared in issue #1241, that was a state diagram and it did draw thicker underlines.   I can search for all the classes that use this path in the code, but I only see the class names and not the packages they are in, so I'm not yet able to recognize which diagrams they belong to.

3. Other entities, such as Classes, follow a slightly different path when determining the hyperlink underline.   They actually look up the value of **hyperlinkunderline** and only care whether or not it is set to "false".  If it can't find **hyperlinkunderline**, it takes that as it not being false.  So two things to note: a) the Class entity doesn't lookup **hyperlinkUnderlineThickness** directly either, so no matter what value is set with that, it doesn't care, even a 0 won't disable the underline on a Class entity header, and b) since it doesn't actually use the value of **hyperlinkunderline**, only checking whether or not it is false, it doesn't increase the thickness on greater values.    There are a number of entities that use this approach, but it may be difficult for me to list them.   For now it might be easier to try different types of diagrams and see which respond.

4. So I'm going to submit the fix to `skimparam hyperlineUnderline true` and the issue of thickness can be a new, separate, issue.

Hopefully this explains it all.  Feel free to ask me if something doesn't make sense.

**Comment by The-Lum** [comment]
Hi all,

_[This is an Issue Review]_ 👀 
This is now fixed on [V1.2024.5](https://github.com/plantuml/plantuml/releases/tag/v1.2024.5).
- Thanks @VladimirAlexiev for the report.
- Thanks PlantUML team and @jimnelson372 for your work.

Regards.
